src/main/java/com/mvplevel1/service/OfferService.java

```diff
@@ -0,0 +1,132 @@
+                    LOGGER.error("term");
+                }
+        }
+        else {
+            LOGGER.error("Please enter a valid name");
+        }
+
+        return loanOffers;
+    }
+
+
+    public CreditDTO offerLoanCalculation(ScoringDataDTO requestDto)
+    {
+
+        BigDecimal monthlyPayment;
+        BigDecimal totalCost;
+
+        CreditDTO response = new CreditDTO();
+        BigDecimal rate = new BigDecimal(10);
+        response.setRate(rate);
+        rate = response.getRate();
+
+
+        if(requestDto.getEmployment().getEmploymentStatus().toString().matches(EmploymentStatus.UNEMPLOYED.name())){
+             LOGGER.error("Sorry you don't qualify for a loan");
+         }
+         if(requestDto.getEmployment().getEmploymentStatus().toString().matches(EmploymentStatus.SELFEMPLOYED.name()))
+         {
+           rate =  rate.add(new BigDecimal(1));
+         }
+         if(requestDto.getEmployment().getEmploymentStatus().toString().matches(EmploymentStatus.BUSINESSOWNER.name())){
+
+             rate = rate.add(new BigDecimal(3));
+         }
+         if(requestDto.getMaritalStatus().toString().matches(MaritalStatus.MARRIED.name())){
+
+             rate = rate.subtract(new BigDecimal(3));
+         }
+         if(requestDto.getMaritalStatus().toString().matches(MaritalStatus.DIVORCED.name()))
+         {
+             rate = rate.subtract(new BigDecimal(1));
+         }
+         if(requestDto.getDependentAmount() > 1)
+         {
+             rate = rate.add(new BigDecimal(1));
+         }
+         if (Period.between(requestDto.getBirthdate() , LocalDate.now()).getYears() < 18 && Period.between(requestDto.getBirthdate() , LocalDate.now()).getYears() > 60)
+         {
+             LOGGER.error("Sorry you dont qualify for loan");
+         }
+         if (requestDto.getGender().toString().matches(Gender.FEMALE.name()) || Period.between(requestDto.getBirthdate() , LocalDate.now()).getYears() == 35 && Period.between(requestDto.getBirthdate() , LocalDate.now()).getYears() > 60)
+         {
+             rate = rate.subtract(new BigDecimal(3));
+         }
+         if (requestDto.getGender().toString().matches(Gender.MALE.name()) && Period.between(requestDto.getBirthdate() , LocalDate.now()).getYears() == 35 && Period.between(requestDto.getBirthdate() , LocalDate.now()).getYears() > 55)
+         {
+             rate = rate.subtract(new BigDecimal(3));
+         }
+        if (requestDto.getEmployment().getWorkExperienceTotal() < 12 && requestDto.getEmployment().getWorkExperienceCurrent() < 3)
+         {
+             LOGGER.error("Sorry you do not qualify for a loan");
+         }
+        totalCost = requestDto.getAmount().multiply(rate);
+        monthlyPayment = totalCost.divide(new BigDecimal(requestDto.getTerm()),  RoundingMode.HALF_UP);
+
+        LOGGER.info("Results " + new CreditDTO(rate, totalCost, monthlyPayment));
+        return new CreditDTO(rate, totalCost, monthlyPayment);
+    }
+
```

Comment:
I don't see calculation of lonan params (monthly payment, payment schedule, total amount, etc.). Will add it?:)